hydrolib/core/io/mdu/models.py

```diff
@@ -820,3 +821,7 @@ def _get_relative_mode_from_data(cls, data: Dict[str, Any]) -> ResolveRelativeMo
             return ResolveRelativeMode.ToAnchor
         else:
             return ResolveRelativeMode.ToParent
+
+    def _serialize(self, _: dict) -> None:
+        config = SerializerConfig(skip_empty_properties=False)
+        write_ini(self._resolved_filepath, self._to_document(), config=config)
```

Comment:
Visual Studio gives the following error for `self._resolved_filepath`:

`Argument of type "Path | None" cannot be assigned to parameter "path" of type "Path" in function "write_ini"
  Type "Path | None" cannot be assigned to type "Path"
    Type "None" cannot be assigned to type "Path"`

I think it is because `_resolved_filepath` is `Optional` and `write_ini()` expects a non-optional path.

Reply:
Is that so? I see such a check only in `ParsableFileModel._serialize()`, but that will not be called if a subclass overrides that method, for example in `INIModel._serialize()`.

I think the place that now makes sure everything runs well is this:
```
class FileModel(BaseModel, ABC):
# ...
    def _save_instance(self) -> None:
        if self.filepath is None:
```

Should we not add documentation that `_serialize()` should always only be called via `save()` and never directly?

Reply:
@priscavdsluis wrote that the latter is implicitly so, because `_serialize()` is a private function. That is true for end-user APIdocs. But amongst core-developers, it is good to know that *other* private functions should probably also not call `_serialize` directly. Maybe some one-liner docs in the function body?

Reply:
I added some docs and will close this PR.